stdlib/toolchain/Compatibility56/include/Concurrency/VoucherShims.h

```diff
@@ -92,8 +95,14 @@ static inline bool voucher_needs_adopt(void * _Nullable voucher) {
 
 static inline bool swift_voucher_needs_adopt(voucher_t _Nullable voucher) {
 #if __APPLE__
-  if (__builtin_available(macOS 12.0, iOS 15.0, tvOS 15.0, watchOS 8.0, *))
-    return voucher_needs_adopt(voucher);
+  // _Z19voucher_needs_adoptP9voucher_s
+  const auto voucherNeedsAdopt =
+      reinterpret_cast<bool(*)(voucher_t)>(SWIFT_LAZY_CONSTANT(
+          dlsym(RTLD_DEFAULT, "_Z19voucher_needs_adoptP9voucher_s")));
```

Comment:
This is an inline function, are we picking up a copy that got inlined into the main concurrency runtime or something? I'm not sure this actually works.

I will note that `return true` is always OK, `false` is just a performance optimization, so worst case we still don't break anything.